lib/pos_consensus.go

```diff
@@ -657,6 +661,20 @@ func (cc *FastHotStuffConsensus) fetchValidatorListsForSafeBlocks(blocks []*MsgD
 	return blocksWithValidatorLists, nil
 }
 
+func (fc *FastHotStuffConsensus) createBlockProducer(bav *UtxoView) (*PosBlockProducer, error) {
+	blockProducerBlsPublicKey := fc.signer.GetPublicKey()
+	blockProducerValidatorEntry, err := bav.GetCurrentSnapshotValidatorBLSPublicKeyPKIDPairEntry(blockProducerBlsPublicKey)
+	if err != nil {
+		return nil, errors.Errorf("Error fetching validator entry for block producer: %v", err)
+	}
+	blockProducerPublicKeyBytes, blockProducerPublicKeyExists := bav.PKIDToPublicKey[*blockProducerValidatorEntry.PKID]
+	if !blockProducerPublicKeyExists {
+		return nil, errors.Errorf("Error fetching public key for block producer: %v", err)
+	}
```

Comment:
```suggestion
	blockProducerPublicKeyBytes := bav.GetPublicKeyForPKID(blockProducerValidatorEntry.PKID)
```
We also want to consider what is in the DB. This function does not return an error. We only store a mapping of public key to PKID (and vice versa) if there was a swap identity performed on the public key. Otherwise the PKID is composed of the same bytes as the public key.